components/rhs_header_post/rhs_header_post.tsx

```diff
@@ -12,27 +11,25 @@ import Constants, {RHSStates} from 'utils/constants';
 import {isMobile} from 'utils/utils.jsx';
 import {browserHistory} from 'utils/browser_history';
 
-export default class RhsHeaderPost extends React.PureComponent {
-    static propTypes = {
-        rootPostId: PropTypes.string.isRequired,
-        channel: PropTypes.object.isRequired,
-        previousRhsState: PropTypes.oneOf(
-            Object.values(RHSStates),
-        ),
-        isExpanded: PropTypes.bool.isRequired,
-        relativeTeamUrl: PropTypes.string.isRequired,
-        actions: PropTypes.shape({
-            setRhsExpanded: PropTypes.func,
-            showMentions: PropTypes.func,
-            showSearchResults: PropTypes.func,
-            showFlaggedPosts: PropTypes.func,
-            showPinnedPosts: PropTypes.func,
-            closeRightHandSide: PropTypes.func,
-            toggleRhsExpanded: PropTypes.func,
-        }),
+interface RhsHeaderPostProps {
+    isExpanded: boolean;
+    rootPostId: string;
+    previousRhsState?: string;
+    relativeTeamUrl: string;
+    channel: {display_name: string};
+    actions: {
+        setRhsExpanded: Function;
```

Comment:
Can we be more specific as to what the functions are here? Otherwise we could technically pass in any function as a prop here, and it kinda circumvents type checking.